newIDE/app/src/UI/Reponsive/ResponsiveWindowMeasurer.js

```diff
@@ -7,41 +7,78 @@ import useOnResize from '../../Utils/UseOnResize';
 // Medium corresponds to tablets and small screens.
 // Large corresponds to most laptop and desktop screens.
 // Xlarge corresponds to large desktop screens.
-export type WidthType = 'small' | 'medium' | 'large' | 'xlarge';
+export type WindowSizeType = 'small' | 'medium' | 'large' | 'xlarge';
 const sizeThresholds = {
   small: 500,
   medium: 1150,
   large: 1500,
 };
 
 type Props = {|
-  children: (width: WidthType) => React.Node,
+  children: ({
+    windowSize: WindowSizeType,
+    isMobile: boolean,
+    isMediumScreen: boolean,
+    isLandscape: boolean,
+  }) => React.Node,
 |};
 
 /**
- * Wraps useResponsiveWindowWidth in a component.
+ * Wraps useResponsiveWindowSize in a component.
  */
 export const ResponsiveWindowMeasurer = ({ children }: Props) =>
-  children(useResponsiveWindowWidth());
+  children(useResponsiveWindowSize());
 
 /**
  * Return the size of the window.
  * This considers a window to be "small" if *both* the width and height
  * are small.
  */
-export const useResponsiveWindowWidth = (): WidthType => {
+export const useResponsiveWindowSize = (): {
+  windowSize: WindowSizeType,
+  isMobile: boolean,
+  isMediumScreen: boolean,
```

Comment:
I guess it's to avoid rewriting this line in the multiple components we do this multiple times, similar to isMobile.
we "just" have to use the boolean coming from the hook.
I agree it's duplicating the information, but it actually cleans up some components